packages/gestalt/src/SideNavigationTopItem.js

```diff
@@ -41,36 +49,38 @@ type Props = {|
   /**
    * Callback when the user selects an item using the mouse or keyboard.
    */
-  onSelect: ({|
+  onClick?: ({|
     event:
       | SyntheticMouseEvent<HTMLDivElement>
       | SyntheticKeyboardEvent<HTMLDivElement>
       | SyntheticMouseEvent<HTMLAnchorElement>
       | SyntheticKeyboardEvent<HTMLAnchorElement>,
-    item: {|
-      label: string,
-      value: string,
-    |},
+    dangerouslyDisableOnNavigation: () => void,
   |}) => void,
   /**
-   * Object detailing the label and value for this item.
+   * Label for the item.
    */
-  item: {| label: string, value: string |},
+  label: string,
 |};
 
 /**
- * Use [SideNavigation.Item](https://gestalt.pinterest.systems/sidenavigation#SideNavigation.Item) for navigation, when the SideNavigation item navigates to a new page or sectoin.
+ * Use [SideNavigation.TopItem](https://gestalt.pinterest.systems/sidenavigation#SideNavigation.TopItem) for navigation, when the SideNavigation.TopItem navigates to a new page or sectoin.
  */
-export default function SideNavigationItem({
+export default function SideNavigationTopItem({
```

Comment:
Can we keep the name of this component as "Item"? For the simplest use cases, it'd be nice to keep usage as something like this
```js
<SideNav>
  <SideNav.Item />
  <SideNav.Item />
  <SideNav.Item />
</SideNav>
```